Adds sorting parameter to series ltitools

jduehring opened the pull request:
This PR adds the option to specify sorting criteria to the series ltitools.

### How to test this patch

Before this PR, it was not possible to set a sorting order in the series ltitools. Calling for example https://stable.opencast.org/ltitools/index.html?subtool=series&sort=title would have no impact to the sorting. 

With the PR you should be able to make use of this parameter, either directly in the url or through the tool configuration of your LMS (as explained in https://docs.opencast.org/r/16.x/admin/#configuration/ltimodule/#specifying-lti-tools). 

I could not test the patch with an LMS yet, so feel free to test, if you have this option. 

### Your pull request should…

* [ ] have a concise title
* [ ] [close an accompanying issue](https://docs.opencast.org/develop/developer/#participate/development-process/#automatically-closing-issues-when-a-pr-is-merged) if one exists
* [ ] [be against the correct branch](https://docs.opencast.org/develop/developer/development-process#acceptance-criteria-for-patches-in-different-versions)
* [ ] include migration scripts and documentation, if appropriate
* [ ] pass automated tests
* [ ] have a clean commit history
* [ ] [have proper commit messages (title and body) for all commits](https://medium.com/@steveamaza/e028865e5791)
* [ ] explain why it needs to be merged into the legacy branch, if it is targeting the legacy branch

ottsimon commented:
Tested with Moodle 4.1  
Sorting works as it should, making it much easier to find a single video in a series of many.

Arnei reviewed (changes requested):
Looks reasonable and works.

Does the weird thing where upper and lowercase strings are sorted separately, but that problem lies with the search service, not this PR.

One thing I would like to see is mention of this new parameter on ltitools/index.html page

jduehring committed:
fix lti-tools-sorting

jduehring committed:
add created sort param + included documentation in ltitools/index.html

jduehring committed:
Improve description of lti url parameter

jduehring commented:
@Arnei I added the information to the  ltitools/index.html page, and by request extended the SearchRestService so we can also sort by the "created" date. 

I also rebased the PR to target r/17.x so that it may be able to make it into OC 17.2.
